### DISCUSSION TITLE
Generating str/int/float/bool values with GPT2 using generate.json leads to absurd results

### DISCUSSION THREAD

**Discussion by gaspard-dv** [outlines-dev/outlines#461] [Weird model output]
### Describe the issue as clearly as possible:

I experimented a bit with outlines and I wonder what would be the recommended way to generate bools/numbers/ints.

* When I use `generate.format(model, <type>, allow_empty_tokens=False)` the resulting response is often just an empty string, which can't be converted to the desired `<type>`.
* When I use `generate.json(model, json.dumps({"type": "<type_in_json_schema_style>"})` the model tends to "block" or to produce stupid answers.

Regardless of my question, I believe that this second bullet point is an issue, so I'm adding a reproduction snippet below.

*NB: I'm not allowed to add the "correctness" label.*

### Steps/code to reproduce the bug:

```python
from json import dumps

from outlines import models
from outlines.text.generate import continuation
from outlines.text.generate import json
import torch

model = models.transformers("gpt2")
prompt = "a b c d e f"

rng = torch.Generator()

rng.manual_seed(789001)
print(continuation(model, max_tokens=30)(prompt, rng=rng))

rng.manual_seed(789001)
print(json(model, dumps({"type": "string"}))(prompt, rng=rng))
```

### Expected result:

```shell
EXPECTED:
g h i j k (...)
g h i j k (...)

ACTUAL:
g h i j k (...)
for all readable words H. Servo Oamber Trident Warrior Drug of others (...)
```

### Outlines/Python version information:

```python
Outlines 0.0.13
Python 3.10.12
```

**Comment by rlouf** [comment]
Will take a look!

**Comment by brandonwillard** [comment]
Just as described in https://github.com/outlines-dev/outlines/discussions/460#discussioncomment-7912276, this should be converted to a discussion.

**Reply by brandonwillard** [reply]
It does look like this one has the potential to involve a real bug, though, so we might end up changing this back...
